src/routers/v1/SearchRouter.js

```diff
@@ -72,22 +74,62 @@ class SearchRouter extends ApplicationRouter<Array<Object>> {
       const autocompleteResult = JSON.parse(autocompleteRequest);
 
       const { predictions } = autocompleteResult;
-      const googlePredictions = predictions.map(p => ({
-        type: GOOGLE_PLACE,
-        detail: p.structured_formatting.secondary_text,
-        name: p.structured_formatting.main_text,
-        placeID: p.place_id,
+
+      const googlePredictions = await Promise.all(predictions.map(async (p): Promise<Object> => {
+        const placeIDCoords = await getCoordsForPlaceID(p.place_id);
+        return {
+          type: GOOGLE_PLACE,
+          detail: p.structured_formatting.secondary_text,
+          name: p.structured_formatting.main_text,
+          lat: placeIDCoords.lat,
+          long: placeIDCoords.long,
+        };
       }));
-      const filteredPredictions = getFilteredPredictions(googlePredictions, formattedStops);
-      cache.set(query, filteredPredictions);
 
-      // Return the list of googlePlaces and busStops
-      return filteredPredictions.concat(formattedStops);
+      if (await googlePredictions) {
```

Comment:
why do we need to `await` here,  `googlePredictions` was already `await`-ed

Reply:
+1